src/hotspot/share/adlc/adlArena.hpp

```diff
@@ -104,9 +104,12 @@ class AdlArena: public AdlCHeapObj {
 
   // Fast allocate in the arena.  Common case is: pointer test + increment.
   void* Amalloc(size_t x) {
+  assert(x <= (size_t)1 << 31, "unreasonable arena allocation size");
```

Comment:
This seems mostly unrelated to the other overflow avoidance checking, and I think doesn't belong in
this PR.  For one thing, this might not be the right mechanism.  Perhaps an Arena should have a
configurable max allocation size?  And shouldn't the max allocation be applied to Arealloc too?